app/controllers/users_controller.rb

```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+before_action :authenticate_user!, :except => [:index, :new]
+
+
```

Comment:
It is one of the requirement to display all users
we should have something like this :point_right:  `@users = User.all`